src/test/java/wolox/training/controllers/BookControllerTest.java

```diff
@@ -0,0 +1,185 @@
+package wolox.training.controllers;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.isA;
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
+
+@WebMvcTest(BookController.class)
+class BookControllerTest {
+
+    @Autowired
+    private MockMvc mvc;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @MockBean
+    private BookRepository bookRepository;
+
+    @Test
+    void whenGetAllBooks_thenReturnJsonArray() throws Exception {
+        //Given
+        Book bookTest = new Book(null,"Mystery","Paolo", "image.png", "Sorcerer's Stone", "-", "Salamander","2000", 250,"123456789");
```

Comment:
You can put the tag @BeforeEach, and accompany it with a setup () method, this method is executed every time you are going to perform a test, in this way, the initialization of objects only takes one line for all the methods. The setup () method looks something like this:
  
public void setUp () {
bookTest = new Book (null, "Mystery", "Paolo", "image.png", "Sorcerer's Stone", "-", "Salamander", "2000", 250, "123456789");
  }